app.js

```diff
@@ -0,0 +1,180 @@
+/*
+ * (C) Copyright 2018 o2r project.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const express = require('express');
+const cookieParser = require('cookie-parser');
+const bodyParser = require('body-parser');
+const errorHandler = require('errorhandler');
+const session = require('express-session');
+const passport = require('passport');
+const routes = require('./lib/routes');
+const debug = require('debug')('guestlister');
+const config = require('./config/config');
+
+const app = express();
+const mongojs = require('mongojs');
+const mongoose = require('mongoose');
+const backoff = require('backoff');
+const MongoStore = require('connect-mongodb-session')(session);
+
+// use ES6 promises for mongoose
+mongoose.Promise = global.Promise;
+
+const dbURI = config.mongo.location + config.mongo.database;
+mongoose.connect(dbURI, {
+    promiseLibrary: global.Promise // use ES6 promises for underlying MongoDB DRIVE
+});
+mongoose.connection.on('error', (err) => {
+    debug('Could not connect to MongoDB @ %s: %s', dbURI, err);
+});
+// If the Node process ends, close the Mongoose connection
+process.on('SIGINT', function () {
+    mongoose.connection.close(function () {
+        debug('Mongoose default connection disconnected through app termination signal (SIGINT)');
+        process.exit(0);
+    });
+});
+
+// Passport configuration
+require('./lib/auth');
+
+function initApp(callback) {
+    debug('Initialize application');
+
+    // Express configuration
+    app.set('view engine', 'ejs');
+    app.use(express.static(__dirname + '/css'));
+    app.use(cookieParser());
+    app.use(bodyParser.json({ extended: false }));
+    app.use(bodyParser.urlencoded({ extended: false }));
+    app.use(errorHandler());
+    // todo check if guestlister has to use same session as bouncer
```

Comment:
The guestlister should _not_ use a session middleware. Please also remove `require('express-session');`